L1Trigger/DemonstratorTools/interface/GTTInterface.h

```diff
@@ -0,0 +1,100 @@
+#ifndef L1Trigger_DemonstratorTools_GTTInterface_h
+#define L1Trigger_DemonstratorTools_GTTInterface_h
+
+#include <cstddef>
+
+#include "L1Trigger/DemonstratorTools/interface/LinkId.h"
+#include "L1Trigger/DemonstratorTools/interface/ChannelSpec.h"
+#include "L1Trigger/DemonstratorTools/interface/EventData.h"
+#include "L1Trigger/DemonstratorTools/interface/FileFormat.h"
+#include "L1Trigger/DemonstratorTools/interface/Frame.h"
+
+namespace l1t::demo {
+  // map of logical channel ID -> [TMUX period, interpacket-gap & offset; channel indices]
+  typedef std::map<LinkId, std::pair<ChannelSpec, std::vector<size_t>>> ChannelMap_t;
+
+  static constexpr size_t kFramesPerTMUXPeriod = 9;
+  static constexpr size_t kGapLengthInput = 6;                //defined in terms of nTracks * (3/2) - 3 * 54?
+  static constexpr size_t kGapLengthOutputToCorrelator = 44;  //can be defined in terms of 54 - nVertices?
+  static constexpr size_t kGapLengthOutputToGlobalTriggerSums = 3;
+  static constexpr size_t kGapLengthOutputToGlobalTriggerTaus = 36;
+  static constexpr size_t kGapLengthOutputToGlobalTriggerMesons = 15;
+  static constexpr size_t kGapLengthOutputToGlobalTriggerVertices = 6;
+  static constexpr size_t kTrackTMUX = 18;  //TMUX of the TrackFindingProcessors
+  static constexpr size_t kGTTBoardTMUX =
+      6;  //TMUX of the GTT in the current configuration: 6 boards running 3 events in parallel, with a paired board running parallel algorithms
+  static constexpr size_t kMaxLinesPerFile = 1024;
+
+  static constexpr size_t kVertexChanIndex = 0;
+
+  // TRACKS from TFP
+  static const std::map<l1t::demo::LinkId, std::vector<size_t>> kChannelIdsInput = {
+      /* logical channel within time slice -> vector of channel indices (one entry per time slice) */
+      /* for first link in a time slice, the channel index is 1 for 1st time slice, channel 19 in the 2nd*/
+      {{"tracks", 0}, {0, 18, 36}},
+      {{"tracks", 1}, {1, 19, 37}},
+      {{"tracks", 2}, {2, 20, 38}},
+      {{"tracks", 3}, {3, 21, 39}},
+      {{"tracks", 4}, {4, 22, 40}},
+      {{"tracks", 5}, {5, 23, 41}},
+      {{"tracks", 6}, {6, 24, 42}},
+      {{"tracks", 7}, {7, 25, 43}},
+      {{"tracks", 8}, {8, 26, 44}},
+      {{"tracks", 9}, {9, 27, 45}},
+      {{"tracks", 10}, {10, 28, 46}},
+      {{"tracks", 11}, {11, 29, 47}},
+      {{"tracks", 12}, {12, 30, 48}},
+      {{"tracks", 13}, {13, 31, 49}},
+      {{"tracks", 14}, {14, 32, 50}},
+      {{"tracks", 15}, {15, 33, 51}},
+      {{"tracks", 16}, {16, 34, 52}},
+      {{"tracks", 17}, {17, 35, 53}}};
+
+  static const ChannelMap_t kChannelMapInput = {
```

Comment:
I think that `kChannelMapInput` is just a combination of the info from `kChannelIdsInput` and `kChannelSpecsInput`. If that's correct, couldn't `kChannelMapInput` be removed (to avoid duplication) by instead using the 2nd `BoardDataReader` constructor on line 95 to 100 of `GTTFileReader.cc`?

Reply:
There's an issue in the method that merges the independent Ids and Specs to form a ChannelMap, so I'm going to leave the current implementation until that is resolved.

Reply:
@NJManganelli Can you clarify what the issue is in that merging method? Sorry if you've already mentioned this over email.